cms/private_media/wagtail_hooks.py

```diff
@@ -0,0 +1,22 @@
+from typing import TYPE_CHECKING
+
+from django.core.exceptions import PermissionDenied
+from wagtail import hooks
+from wagtail.documents.permissions import permission_policy
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+
+    from cms.private_media.models import PrivateDocumentMixin
+
+
+@hooks.register("before_serve_document")
+def protect_private_documents(document: "PrivateDocumentMixin", request: "HttpRequest") -> None:
+    """Block access to private documents if the user has insufficient permissions."""
+    if document.is_private and (
+        not request.user.is_authenticated
+        or not permission_policy.user_has_any_permission_for_instance(
+            request.user, ["choose", "add", "change"], document
+        )
+    ):
```

Comment:
As above, we need to be able to serve even if it is private just incase ACL set failed.

Reply:
Documents are different to images in thar they are ALWAYS served by the view, regardless of privacy values. So, in essence, the ACL is irrelevant here, as the storage backend uses its credentials to stream the file